### PR TITLE
docs: Assorted cleanups of links and `literals`

### PR DESCRIPTION
**Pull request by phlax**
Cleanup of links/`literals` that are using either single backticks or italics - that were ~easy to find and some adjacent ones

Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
changes the following files:

```console
docs/root/api-docs/xds_protocol.rst
docs/root/configuration/http/http_conn_man/stats.rst
docs/root/configuration/http/http_conn_man/vhds.rst
docs/root/configuration/http/http_filters/buffer_filter.rst
docs/root/configuration/http/http_filters/compressor_filter.rst
docs/root/configuration/http/http_filters/file_system_buffer_filter.rst
docs/root/configuration/http/http_filters/grpc_http1_bridge_filter.rst
docs/root/configuration/http/http_filters/jwt_authn_filter.rst
docs/root/configuration/http/http_filters/oauth2_filter.rst
docs/root/configuration/http/http_filters/stateful_session_filter.rst
docs/root/configuration/http/http_filters/sxg_filter.rst
docs/root/configuration/operations/config_validation/config_validation.rst
docs/root/configuration/operations/overload_manager/overload_manager.rst
docs/root/configuration/other_features/vcl.rst
docs/root/configuration/security/secret.rst
docs/root/configuration/upstream/cluster_manager/cluster_runtime.rst
docs/root/faq/configuration/sni.rst
docs/root/faq/configuration/timeouts.rst
docs/root/faq/debugging/why_is_envoy_sending_http2_resets.rst
docs/root/faq/load_balancing/region_failover.rst
docs/root/intro/arch_overview/http/http3.rst
docs/root/intro/arch_overview/http/http_routing.rst
docs/root/intro/arch_overview/operations/hot_restart.rst
docs/root/intro/arch_overview/other_protocols/redis.rst
docs/root/operations/admin.rst
docs/root/operations/tools/schema_validator_check_tool.rst
docs/root/start/install.rst
docs/root/start/sandboxes/dynamic-configuration-filesystem.rst
```

File: docs/root/api-docs/xds_protocol.rst [hunk 1]
```diff
@@ -466,24 +466,24 @@ resource type have had an empty :ref:`resource_names <envoy_v3_api_field_service
...
-- Client sends a request with :ref:`resource_names_subscribe <envoy_v3_api_field_service.discovery.v3.DeltaDiscoveryRequest.resource_names_subscribe>` set to "A". Server interprets this as continuing the existing subscription to "*" and adding a new subscription to "A".
-- Client sends a request with :ref:`resource_names_unsubscribe <envoy_v3_api_field_service.discovery.v3.DeltaDiscoveryRequest.resource_names_unsubscribe>` set to "*". Server interprets this as unsubscribing to "*" and continuing the existing subscription to "A".
+- Client sends a request with :ref:`resource_names_subscribe <envoy_v3_api_field_service.discovery.v3.DeltaDiscoveryRequest.resource_names_subscribe>` unset. Server interprets this as a subscription to ``*``.
+- Client sends a request with :ref:`resource_names_subscribe <envoy_v3_api_field_service.discovery.v3.DeltaDiscoveryRequest.resource_names_subscribe>` set to "A". Server interprets this as continuing the existing subscription to ``*`` and adding a new subscription to "A".
```
Comment (adisuissa) [inline docs/root/api-docs/xds_protocol.rst:485]: OOC why do different quotes are used for * and for A?
(I've seen the [rendered docs](https://storage.googleapis.com/envoy-pr/4221e3b/docs/api-docs/xds_protocol.html#how-the-client-specifies-what-resources-to-return) just wondering when I should use the 2 types of quotes).

File: docs/root/configuration/http/http_conn_man/stats.rst [hunk 2]
```diff
@@ -3,121 +3,121 @@
...
+   ``downstream_cx_destroy_remote_active_rq``, Counter, Total connections destroyed remotely with 1+ active requests
+   ``downstream_rq_total``, Counter, Total requests
 
 .. _config_http_conn_man_stats_per_listener:
 
 Per listener statistics
 -----------------------
 
-Per listener statistics are rooted at *listener.<address>*.
+Per listener statistics are rooted at ``listener.<address>``.
```
Comment (adisuissa) [inline docs/root/configuration/http/http_conn_man/stats.rst:91]: nit:
```suggestion
Per listener statistics are rooted at ``listener.<address>.``
```

File: docs/root/faq/configuration/timeouts.rst [hunk 3]
```diff
@@ -28,7 +28,7 @@ Connection timeouts apply to the entire HTTP connection and all streams the conn
   <envoy_v3_api_field_extensions.filters.network.http_connection_manager.v3.HttpConnectionManager.common_http_protocol_options>`
   field in the HTTP connection manager configuration. To modify the idle timeout for upstream
   connections use the
-  :ref:`common_http_protocol_options <envoy_v3_api_field_extensions.upstreams.http.v3.HttpProtocolOptions.common_http_protocol_options>` field in the Cluster's :ref:`extension_protocol_options<envoy_v3_api_field_config.cluster.v3.Cluster.typed_extension_protocol_options>`, keyed by `envoy.extensions.upstreams.http.v3.HttpProtocolOptions`
+  :ref:`common_http_protocol_options <envoy_v3_api_field_extensions.upstreams.http.v3.HttpProtocolOptions.common_http_protocol_options>` field in the Cluster's :ref:`extension_protocol_options<envoy_v3_api_field_config.cluster.v3.Cluster.typed_extension_protocol_options>`, keyed by :ref:`HttpProtocolOptions <envoy_v3_api_msg_extensions.transport_sockets.tls.v3.UpstreamTlsContext>`
```
Comment (adisuissa) [inline docs/root/faq/configuration/timeouts.rst:31]: Can you double check that the link to `HttpProtocolOptions` is the correct one?

**Review by adisuissa — COMMENTED** [review]
Thanks! Left some minor comments.

File: docs/root/api-docs/xds_protocol.rst (hunk 1 above)
Comment (phlax) [reply docs/root/api-docs/xds_protocol.rst:485]: i think in this case its a literal - ie should use ` `` `

the only time we want to use `*text*` is for emphasis - never for literals

agreed that this file has been left a bit inconsistent - ill update it now to use literals

File: docs/root/configuration/http/http_conn_man/stats.rst (hunk 2 above)
Comment (phlax) [reply docs/root/configuration/http/http_conn_man/stats.rst:91]: i think it needs both - the `.` in the stats label and a full stop

Comment (adisuissa) [reply docs/root/configuration/http/http_conn_man/stats.rst]: FWIW, I've seen this in other places around the docs (e.g., lines 133, 135 of this file).
Not sure if it is worth updating all the places, but if we can have some clear guideline, we can gradually move the docs there.

File: docs/root/faq/configuration/timeouts.rst (hunk 3 above)
Comment (phlax) [reply docs/root/faq/configuration/timeouts.rst]: fixed

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

Not sure why CI reports the error, but maybe merge-main will solve this.

**Pull request merged**